### PR TITLE
Oat

### PR DESCRIPTION
**Pull request by adelmemariani**

### CODE REVIEW

**Comment by jh-RLI** [comment]
Hi @adelmemariani Thanks for your work - I don't know why there are two new migrations in dataedit, you remove fields and then add them back. 
image

Also, this PR again includes all the work for the OAT and your work on the factsheet. But if you think you can't implement the changes independently, then just go ahead with this PR. I think at this point we could have implemented the code from scratch and it would have been easier for you :D 

I still think you've added too many changes (stuff that is already in dev and will be overwritten by this PR) and I'm still not sure if you'll overwrite existing code, but we've discussed this several times - I'll try to help you with the CI error, but I don't have the time to make sure this PR doesn't break any of the existing code.

File: dataedit/static/metaedit/metaedit.js [hunk 1]
```diff
@@ -2,363 +2,412 @@
...
+
+                // all done
+                $('#metaedit-loading').addClass('d-none');
+                $('#metaedit-icon').removeClass('d-none');
+                $('#metaedit-controls').removeClass('d-none');
+
+                // TODO catch init error
+
+
+                window.JSONEditor.defaults.callbacks = {
```
Comment (jh-RLI) [inline dataedit/static/metaedit/metaedit.js:315]: This is your code 👍

File: dataedit/static/metaedit/metaedit.js [hunk 2]
```diff
@@ -2,363 +2,412 @@
...
+                "autocomplete": {
+                    "search_name": function search(jseditor_editor, input) {
+
+                      var url = "/api/v0/oeo-search?query=" + input
```
Comment (jh-RLI) [inline dataedit/static/metaedit/metaedit.js:319]: same as above we should import the path via the settings - you can use the context of the metaEdit view to get this string from django oeplatform/settings.py

File: dataedit/static/metaedit/metaedit.js (hunk 1 above)
Comment (jh-RLI) [reply dataedit/static/metaedit/metaedit.js:315]: We have to add this code to the else in block starting in line 356 - the metaEditor can be initialized as so called "standalone" or via the "edit" button for each table - bot should be able to use the OAT. Maybe we can create a function to aviode more duplicated code?

File: dataedit/static/metaedit/metaedit.js [hunk 3]
```diff
@@ -2,363 +2,412 @@
...
+                        return selected_value;
+                    }
+                }
+              };
+
+
+
+            });
+
+        } else {
```
Comment (jh-RLI) [inline dataedit/static/metaedit/metaedit.js:356]: THis is not your code please dont override existing code

File: dataedit/templates/dataedit/filter.html [hunk 4]
```diff
@@ -80,7 +80,15 @@ <h5>Tags</h5>
 
 {% block after-body-bottom-js %}
 <!-- Latest compiled and minified JavaScript -->
+<<<<<<< HEAD
```
Comment (jh-RLI) [inline dataedit/templates/dataedit/filter.html:83]: this should not be here

File: dataedit/templates/dataedit/filter.html [hunk 5]
```diff
@@ -103,12 +111,15 @@ <h5>Tags</h5>
     $('#search-filter').on('blur', function (e) {
         $("#form-filter").submit();
     });
+<<<<<<< HEAD
```
Comment (jh-RLI) [inline dataedit/templates/dataedit/filter.html:114]: this should not be here

File: dataedit/templates/dataedit/filter.html [hunk 6]
```diff
@@ -80,7 +80,15 @@ <h5>Tags</h5>
...
 <!-- Latest compiled and minified JavaScript -->
+<<<<<<< HEAD
 <script src="{% static 'js/bootstrap-select.min.js' %}"></script><!-- src: https://github.com/snapappointments/bootstrap-select/archive/v1.14.0-beta2.zip -->
+=======
+<script src="{% static 'dataedit/bootstrap-select.min.js' %}"></script><!-- src: https://cdn.jsdelivr.net/npm/bootstrap-select@1.13.0/dist/js/bootstrap-select.min.js -->
+
+
+
+
+>>>>>>> loep_service
```
Comment (jh-RLI) [inline dataedit/templates/dataedit/filter.html:91]: same

File: dataedit/templates/dataedit/filter.html [hunk 7]
```diff
@@ -103,12 +111,15 @@ <h5>Tags</h5>
...
     });
+<<<<<<< HEAD
     /* also update filter on key=Enter in search field */
-    $('#search-filter').on('keydown', function (e) {                    
+    $('#search-filter').on('keydown', function (e) {
         if (e.keyCode == 13) {
             $("#form-filter").submit();
         }
     });
+=======
```
Comment (jh-RLI) [inline dataedit/templates/dataedit/filter.html:121]: same

File: oeplatform/settings.py [hunk 8]
```diff
@@ -88,6 +89,15 @@
     }
 ]
 
+CORS_ORIGIN_WHITELIST = [
```
Comment (jh-RLI) [inline oeplatform/settings.py:92]: this is not used anymore, isnt it? please make sure you dont leave deprecated code here :) Ignore if im wrong

File: oeplatform/urls.py [hunk 9]
```diff
@@ -32,8 +32,5 @@
     url(r"^ontology/", include("ontology.urls")),
     url(r"^tutorials/", include("tutorials.urls")),
     url(r"^viewer/oeo/", include("oeo_viewer.urls")),
-    url(
-        r"^tutorials/[/\w_\d]*/images/(?P<f>[\w_\d]+(\.[\w_\d]+)?)$",
-        ImagesView.as_view(),
-    ),
+    url(r"^tutorials/[/\w_\d]*/images/(?P<f>[\w_\d]+(\.[\w_\d]+)?)$", ImagesView.as_view()),
```
[the remark this answers is not in the document]
Comment (jh-RLI) [reply oeplatform/urls.py:35]: Seems like this is not your code

**Comment by jh-RLI** [comment]
One more thing - I know this feels like a pain in the ***, but please use our contribution guidelines - branches should be called `feature/branch-name#RelatedIssueNumber`.
we work as a team and you are part of the team :) so it's better to keep some structure to make our workflow more reliable and transparent for others - branch name is a small part but quite important

**Comment by jh-RLI** [comment]
I found the reason for the CI error, I updated the omi compiler because it inserted unexpected data in the metadata string. Since this was just released this error was not noticed yet :) . I will try to fix it - It is triggered because the test that fails is trying to compile the metadata json object with a minimum of keys (only the "id" key is present) - you can ignore this for now, I will fix it in OMI. Please have a look at my comments above.

**Comment by adelmemariani** [comment]
> Also, this PR again includes all the work for the OAT and your work on the factsheet.

As it is shown and communicated via emails, I removed the factsheet from the code and the PR does not include the factsheet:

[remove factsheet code](https://github.com/OpenEnergyPlatform/oeplatform/pull/1091/commits/8fb36dd67838d0e4b4167ebcc1f049a9c26188f5)
[8fb36dd](https://github.com/OpenEnergyPlatform/oeplatform/pull/1091/commits/8fb36dd67838d0e4b4167ebcc1f049a9c26188f5)

 but the commit messages are still shown.

**Comment by adelmemariani** [comment]
> I still think you've added too many changes (stuff that is already in dev and will be overwritten by this PR)

The PR only contains the changes for OAT. I just pulled from DEV and checkout a new branch called OAT and added the codes for OAT.

**Comment by adelmemariani** [comment]
> It is triggered because the test that fails is trying to compile the metadata json object with a minimum of keys (only the "id" key is present)

This prevented me from merging my branch and it is not related to the new code and other highlighted issues, I explained it via email.

**Comment by jh-RLI** [comment]
I never blamed you for the CI error, but that was not the main reason blocking the merge. I spent some time reviewing this and it contains several things that we should not include in the codebase.

**Comment by jh-RLI** [comment]
> The PR only contains the changes for OAT. I just pulled from DEV and checkout a new branch called OAT and added the codes for OAT.

If thats the case why are these chainges included in this PR:
image

**Comment by adelmemariani** [comment]
> If thats the case why are these changes included in this PR:

Those are compiled js files for the oeo_viewer. Each release has newly compiled files. These files are the result of running the `npm build` command of the oeo-viewer.  The .gitignore contains 'static', and I have no idea why the 'static' is not ignored. I am currently a bit occupied by other tasks and I can not investigate more. I think we are good to go as all tests are now successful. I would suggest merging the PR.

**Comment by jh-RLI** [comment]
Oaky I will merge the PR soon - just removing your prints and some other things. I'm just saying that the next PR will have to fix the mistakes here - please work on this as soon as your workload is not as it is at the moment :) . And please no more big PR in the future, small prs with eraly PR so others can have a look - qick and dirty is not a good practice to maintain a project in the long term.

**Comment by jh-RLI** [comment]
@adelmemariani can you tell me if this "CORS_ORIGIN_WHITELIST" is needed? you put it in the settings.py

**Review by jh-RLI — APPROVED** [review]
I will approve for now - look at the issues, i will list open tasks there soon.

**Comment by jh-RLI** [comment]
@adelmemariani i cant test this on my local steup :/ or is it not working ATM?
- > 
image

**Comment by adelmemariani** [comment]
> @adelmemariani i cant test this on my local steup :/ or is it not working ATM?
> 
> * >
> 
> image
> 
> image

As you are running it locally, is the leop server running on your local machine too? the image shows a request from your local host.

**Comment by adelmemariani** [comment]
> > @adelmemariani i cant test this on my local steup :/ or is it not working ATM?
> > 
> > * >
> > 
> > image
> > image
> 
> As you are running it locally, is the leop server running on your local machine too? the image shows a request from your local host.

**Comment by adelmemariani** [comment]
As you are running it locally, is the leop server running on your local machine too? the image shows a request from your local host. look at the code here:

https://github.com/OpenEnergyPlatform/oeplatform/blob/oat/api/views.py#L1183
the API requests from the loep server. The production server will communicate with loep server. In my local machine which is in the same network as the loep server, things are working.

**Comment by jh-RLI** [comment]
Yes that's what I thought :) as long as all runs with you it's good. I don't have the loep running locally - thought maybe my local oep communicates with the loep in magdeburg.

**Pull request merged**